content/blog/package-react-components/index.md

```diff
@@ -0,0 +1,776 @@
+---
+title: Package, Publish and Distribute React Components in NPM
+date: "2020-01-18"
+author: franleplant
+description:
+  "In this post we explore in detail how to share React Components which are typically made of a
+  bundle of Javascript, CSS (styles), Images, etc.
+  We explore how to package and bundle the code with Webpack so that it can be consumed appropriately,
+  and we explore how to publish and distribute it to NPM so that it can be easily accessible either
+  via the public NPM repository or a private repository.
+  We also analyse different theming strategies for larger Component Libraries and UI Frameworks.
+  And finally we provide a recommended way of packaging your Components in a simple and very flexible way."
+tags:
+  - React
+  - Javascript
+  - Typescript
+  - Css
+  - Webpack
+  - NPM
+  - code sharing
+  - publishing
+---
+
+## Table of Contents
+
+```toc
+# This code block gets replaced with the TOC
+to-heading: 3
+```
+
+## Introduction
+
+> Check [TLDR](#tldr-how-to-package-and-publish-front-end-libraries-in-2020) if you don't want all
+> details about how and why we reach to the conclusion that that is a valid and desirable solution for the
+> problem at hand
+
+There are diverse situation in which we want to create an NPM package that contains
+a [React][1] Component with css and images.
+
+This situation is particularly common inside companies that want to share components,
+and pieces of code in general, between apps usually inside a private NPM repository, but
+it also apply to Open Source software like a single Component package such as a Date Picker
+or a Multi Select and also to Component Libraries or UI Frameworks like [React Bootstrap][9],
```

Comment:
I think the idea here is to say that it happens for components/pieces of code related to those frameworks instead of the frameworks themselves which was my initial impression when reading.

Reply:
your initial impression is correct, this situation applies to how those frameworks are built. Will try to clarify it.